### PR TITLE
chore(context): update_tags only adds tags with new keys

### PR DESCRIPTION
**Pull request by ZStriker19**
## Description
In the TraceTagsProcessor we run `ctx._update_tags(chunk_root)` which would previously take tags from context and add them onto the span. However this was the case even if the span already had a value. With this change, if a key is already present on the span, then we don't update its value.

Benchmark tests show that this doesn't seem to have an impact: `scripts/perf-run-scenario span Datadog/dd-trace-py@1.x Datadog/dd-trace-py@chore/update_TraceTagsProcessor ./artifacts/  `
README rst — dd-trace-py 2022-08-09 at 10 08 10 AM

## Checklist
- [ ] Title must conform to [conventional commit](https://github.com/conventional-changelog/commitlint/tree/master/%40commitlint/config-conventional).
- [ ] Add additional sections for `feat` and `fix` pull requests.
- [ ] Ensure tests are passing for affected code.
- [ ] [Library documentation](https://github.com/DataDog/dd-trace-py/tree/1.x/docs) and/or [Datadog's documentation site](https://github.com/DataDog/documentation/) is updated. Link to doc PR in description.

## Reviewer Checklist
- [ ] Title is accurate.
- [ ] Description motivates each change.
- [ ] No unnecessary changes were introduced in this PR.
- [ ] PR cannot be broken up into smaller PRs.
- [ ] Avoid breaking [API](https://ddtrace.readthedocs.io/en/stable/versioning.html#interfaces) changes unless absolutely necessary.
- [ ] Tests provided or description of manual testing performed is included in the code or PR.
- [ ] Release note has been added for fixes and features, or else `changelog/no-changelog` label added.
- [ ] All relevant GitHub issues are correctly linked.
- [ ] Backports are identified and tagged with Mergifyio.
- [ ] Add to milestone.

### CODE REVIEW

File: tests/integration/test_integration_snapshots.py [hunk 1]
```diff
@@ -273,3 +275,13 @@ def test_trace_with_wrong_metrics_types_not_sent(metrics):
                 with tracer.trace("child") as child:
                     child._metrics = metrics
         log.exception.assert_called_once_with("error closing trace")
+
+
+@snapshot()
+def test_tracetagsprocessor_only_adds_new_tags():
+    tracer = Tracer()
+    with tracer.trace(name="web.request") as span:
```
Comment (Kyle-Verhoog) [inline tests/integration/test_integration_snapshots.py:283]: should we be doing a `span.context.sampling_priority = <something that isn't user keep>`?

Comment (brettlangdon) [reply tests/integration/test_integration_snapshots.py:283]: `set_tag("manual.drop")` should handle that for us, but I do like the explicit `span.context.sampling_priority = AUTO_KEEP` or something.

**Review by brettlangdon — COMMENTED** [review]
it would be worth running the `span` benchmark scenario for this change.

File: tests/integration/test_integration_snapshots.py (hunk 1 above)
Comment (Kyle-Verhoog) [reply tests/integration/test_integration_snapshots.py:283]: ah right, completely forgot lol..

**Review by mabdinur — APPROVED** [review]

**Comment by majorgreys** [comment]
@Mergifyio backport 1.4

**Pull request merged**